components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/oauth2/OAuthInboundFunctions.java

```diff
@@ -59,6 +59,7 @@ public static InboundAuthenticationRequestConfig putOAuthInbound(ServiceProvider
                     throw buildBadRequestError("Invalid ClientSecret provided for update.");
                 }
 
+                oidcConfigModel.setClientName(application.getApplicationName());
```

Comment:
Shouldn't this be done for both existing and new clients?

Reply:
ie. regardless of an update or create

Reply:
for create , it is done here https://github.com/wso2/identity-api-server/pull/131/files#diff-5a6387cc0923d1a579afa6879c37593aR54